goldmane/pkg/emitter/emitter.go

```diff
@@ -92,6 +115,31 @@ func NewEmitter(opts ...Option) *Emitter {
 	return e
 }
 
+func (e *Emitter) Enable() {
+	close(e.enabled)
+	logrus.Info("Emitter enabled.")
+}
+
+func (e *Emitter) Disable() {
+	e.enabledLock.Lock()
+	defer e.enabledLock.Unlock()
+	e.enabled = make(chan struct{})
+}
+
+func (e *Emitter) isEnabled() bool {
+	e.enabledLock.Lock()
```

Comment:
not actually sure I need this locking... but we do potentially reassign `e.enabled` so it seems right...